src/core.rs

```diff
@@ -730,20 +995,64 @@ pub fn wrap_optimal_fit<'a, T: Fragment, F: Fn(usize) -> usize>(
         cost
     });
 
-    let mut lines = Vec::with_capacity(line_numbers.get(fragments.len(), &minima));
-    let mut pos = fragments.len();
-    loop {
-        let prev = minima[pos].0;
-        lines.push(&fragments[prev..pos]);
-        pos = prev;
-        if pos == 0 {
-            break;
+    let mut line_start = minima[fragments.len()].0;
+
+    WrapOptimalFit {
+        fragments: fragments
+            .into_iter()
+            .enumerate()
+            .rev()
+            .map(|(i, data)| {
+                let eol = i + 1 == line_start;
+                if eol {
+                    line_start = minima[line_start].0;
+                }
+                (data.fragment, eol)
+            })
+            .collect(),
+    }
+}
```

Comment:
Is the `WrapOptimalFit` struct needed here? It seems to simply contain a list of tuples and I guess another `.map` here could make it unnecessary?

Reply:
The key thing is that `WrapOptimalFit::next` calls `.pop()` on the underlying vector, which means that it reverses the order of iteration (which is essential for the algorithm to work). To still support getting the vector directly with `Vec::reverse` like we had before `WrapOptimalFit` also has a `.into_vec()` method.